From 98480e321ffd4975932a3d473a9a412e1bf079d9 Mon Sep 17 00:00:00 2001
From: Zoltan Fridrich <zfridric@redhat.com>
Date: Wed, 15 Sep 2021 10:32:15 +0200
Subject: [PATCH] Fix IPC files override

Fix for a bug where not all relevant IPC files are being loaded.
ie. if there is a user bob who is member of a group bob and there are IPC
files for both user bob and group bob. Only IPC file for user bob would
get loaded.
---
 src/Library/IPCServerPrivate.cpp | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/Library/IPCServerPrivate.cpp b/src/Library/IPCServerPrivate.cpp
index d952350..ce28428 100644
--- a/src/Library/IPCServerPrivate.cpp
+++ b/src/Library/IPCServerPrivate.cpp
@@ -571,10 +571,12 @@ namespace usbguard
   bool IPCServerPrivate::authenticateIPCConnectionDAC(uid_t uid, gid_t gid, IPCServer::AccessControl* const ac_ptr) const
   {
     USBGUARD_LOG(Trace) << "uid=" << uid << " gid=" << gid << " ac_ptr=" << ac_ptr;
-    return \
-      matchACLByUID(uid, ac_ptr) || \
-      matchACLByGID(gid, ac_ptr) || \
-      matchACLByName(uid, gid, ac_ptr);
+
+    bool matched_uid = matchACLByUID(uid, ac_ptr);
+    bool matched_gid = matchACLByGID(gid, ac_ptr);
+    bool matched_name = matchACLByName(uid, gid, ac_ptr);
+
+    return matched_uid || matched_gid || matched_name;
   }
 
   bool IPCServerPrivate::matchACLByUID(uid_t uid, IPCServer::AccessControl* const ac_ptr) const
